src/frontend/org/voltdb/utils/CatalogUtil.java

```
@@ -2256,7 +2263,35 @@ public static CatalogAndDeployment getCatalogFromZK(ZooKeeper zk)
         catalogDeploymentBytes.get(deploymentBytes);
         catalogDeploymentBytes = null;
 
-        return new CatalogAndDeployment(genId, catalogHash, deploymentHash, catalogBytes, deploymentBytes);
+        return new CatalogAndDeployment(genId, catalogHash, catalogBytes, deploymentBytes);
+    }
+
+    public static void copyCurrentCatalogToPreviousZK(ZooKeeper zk, long genId,
+            byte[] catalogBytes, byte[] catalogHash, String deployment) {
+        try {
+            // read the current catalog bytes
+            byte[] data = zk.getData(VoltZK.catalogbytes, false, null);
+            // write to the previous catalog bytes place holder
+            zk.setData(VoltZK.catalogbytesPrevious, data, -1);
+        } catch (KeeperException | InterruptedException e) {
+            VoltDB.crashLocalVoltDB("error read catalog bytes from zookeeper", true, e);
+        }
+    }
+
+    /**
+     * Copy the previous catalog bytes from ZK to the current catalog ZK node
+     */
+    public static void copyPreviousCatalogToCurrentZK(ZooKeeper zk)
+    {
+        try {
+            // read the current catalog bytes
+            byte[] data = zk.getData(VoltZK.catalogbytesPrevious, false, null);
+            assert(data != null);
+            // write to the previous catalog bytes place holder
+            zk.setData(VoltZK.catalogbytes, data, -1);
+        } catch (KeeperException | InterruptedException e) {
+            VoltDB.crashLocalVoltDB("error read catalog bytes from zookeeper", true, e);
```

that was my mistake. In crashLocalVoltDB it calls System.exit(-1), so it should throw an exception first that can be caught by finally block.